ui/core/components/suggest_reforges_action.tsx

```diff
@@ -540,6 +555,49 @@ export class ReforgeOptimizer {
 		);
 	}
 
+	setStatCaps(eventID: EventID, newStatCaps: Stats) {
+		this._statCaps = newStatCaps;
+		this.statCapsChangeEmitter.emit(eventID);
+	}
+
+	get statCaps() {
+		return this.useCustomEPValues ? this._statCaps : this.defaults.statCaps || new Stats();
+	}
+
+	setUseCustomEPValues(eventID: EventID, newUseCustomEPValues: boolean) {
+		if (newUseCustomEPValues !== this.useCustomEPValues) {
+			this.useCustomEPValues = newUseCustomEPValues;
+			this.useCustomEPValuesChangeEmitter.emit(eventID);
+		}
+	}
```

Comment:
I think we should automatically turn off soft cap breakpoints in here if `newUseCustomEPValues` is true and the player's current EP settings do not match any of the EP presets. The reason is that the configured default post-cap EP values will no longer make sense if the user fiddles with custom pre-cap values themselves. Likewise, we should add a listener for EP changes that also turns off soft cap breakpoints if `this.useCustomEPValues` is true at the time when the EP is changed to a non-preset custom setup.